# This is not upstream that I (aacid@kde.org) could find
--- a/src/plugins/scenegraph/d3d12/qsgd3d12engine.cpp	2022-01-25 13:15:40.001262900 +0100
+++ a/src/plugins/scenegraph/d3d12/qsgd3d12engine.cpp	2022-01-25 13:16:42.486380800 +0100
@@ -45,6 +45,7 @@
 #include <QLoggingCategory>
 #include <qmath.h>
 #include <qalgorithms.h>
+#include <d3d12sdklayers.h>
 
 // Comment out to disable DeviceLossTester functionality in order to reduce
 // code size and improve startup perf a tiny bit.
@@ -221,7 +222,7 @@
         if (SUCCEEDED(factory->EnumAdapters1(adapterIndex, &adapter))) {
             adapter->GetDesc1(&desc);
             const QString name = QString::fromUtf16((char16_t *) desc.Description);
-            HRESULT hr = D3D12CreateDevice(adapter.Get(), fl, _uuidof(ID3D12Device), nullptr);
+            HRESULT hr = D3D12CreateDevice(adapter.Get(), fl, __uuidof(ID3D12Device), nullptr);
             if (SUCCEEDED(hr)) {
                 qCDebug(QSG_LOG_INFO_GENERAL, "Using requested adapter '%s'", qPrintable(name));
                 *outAdapter = adapter.Detach();
@@ -238,7 +239,7 @@
         if (desc.Flags & DXGI_ADAPTER_FLAG_SOFTWARE)
             continue;
 
-        if (SUCCEEDED(D3D12CreateDevice(adapter.Get(), fl, _uuidof(ID3D12Device), nullptr))) {
+        if (SUCCEEDED(D3D12CreateDevice(adapter.Get(), fl, __uuidof(ID3D12Device), nullptr))) {
             const QString name = QString::fromUtf16((char16_t *) desc.Description);
             qCDebug(QSG_LOG_INFO_GENERAL, "Using adapter '%s'", qPrintable(name));
             break;
